### PR TITLE
Extract attributes actions to a library that can be shared

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: internal/processor/attraction/attraction.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+	// It uses the attribute specified by `key' to extract values from
+	// The target keys are inferred based on the names of the matcher groups
+	// provided and the names will be inferred based on the values of the
+	// matcher group.
+	// Note: All subexpressions must have a name.
+	// Note: The value type of the source key must be a string. If it isn't,
+	// no extraction will occur.
+	RegexPattern string `mapstructure:"pattern"`
+
+	// FromAttribute specifies the attribute from the span to use to populate
```
Comment (dmitryax) [inline internal/processor/attraction/attraction.go:54]:
```suggestion
	// FromAttribute specifies the attribute to use to populate
```

File: internal/processor/attraction/attraction.go [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+	// the value. If the attribute doesn't exist, no action is performed.
+	FromAttribute string `mapstructure:"from_attribute"`
+
+	// Action specifies the type of action to perform.
+	// The set of values are {INSERT, UPDATE, UPSERT, DELETE, HASH}.
+	// Both lower case and upper case are supported.
+	// INSERT -  Inserts the key/value to spans when the key does not exist.
```
Comment (dmitryax) [inline internal/processor/attraction/attraction.go:61]:
```suggestion
	// INSERT -  Inserts the key/value to attributes when the key does not exist.
```

File: internal/processor/attraction/attraction.go [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+	//           No action is applied to spans where the key already exists.
```
Comment (dmitryax) [inline internal/processor/attraction/attraction.go:62]: should we replace all the `spans` wording to `attributes`?

File: internal/processor/attraction/attraction.go (hunk 1 above)
Comment (bogdandrutu) [reply internal/processor/attraction/attraction.go:54]: Done.

File: internal/processor/attraction/attraction.go (hunk 2 above)
Comment (bogdandrutu) [reply internal/processor/attraction/attraction.go:61]: Done.

File: internal/processor/attraction/attraction.go (hunk 3 above)
Comment (bogdandrutu) [reply internal/processor/attraction/attraction.go:62]: Done.

**Pull request merged**